.github/workflows/release.yml

```diff
@@ -16,33 +16,121 @@
 # specific language governing permissions and limitations
 # under the License.
 #
-name: Release Apache Spark
+
+# This workflow is intended for use in forked repositories
+# when manually dispatching this to create an RC.
+# To enable full release functionality, developers should manually configure
+# the following GitHub Secrets in their repository settings:
+#
+# - ASF_USERNAME:
+#     Your Apache Software Foundation (ASF) account ID.
+#
+# - ASF_PASSWORD:
+#     The password associated with your ASF account.
+#
+# - GPG_PRIVATE_KEY:
+#     Your GPG private key, exported using:
+#       gpg --armor --export-secret-keys ABCD1234 > private.key
+#     Ensure this key is registered with a public key server.
+#     For more details, refer to:
+#       https://spark.apache.org/release-process.html#preparing-gpg-key
+#
+# - GPG_PASSPHRASE:
+#     The passphrase for your GPG private key.
+#
+# This workflow supports dry runs by default. If the required GitHub Secrets are not provided,
+# only dry runs will be executed.
+#
+# In case something goes wrong during the process and a release candidate (RC) needs to be
+# cleaned up, follow these steps:
+#
+# 1. Revert the RC-related commits, such as:
```

Comment:
I usually don't do it as it messes up the commit history further. It's ok to cut the same RC twice, the RC-related commits do not pollute the codebase.